src/web/context/GlobalContext.tsx

```diff
@@ -0,0 +1,231 @@
+
+        case 'CLICKED_HUB_ACCORDION':
+            return handleClickedHubAccordion(mutableState, action.hubAccordionName)
+
+        default:
+            return state
+    }
+}
+
+/**
+ * Adds the client ID to state
+ * 
+ * @param {GlobalContextType} mutableState State object ref for making modifications
+ * @param {string} clientID ID associated with the client session
+ * @returns {GlobalContextType} Updated mutable state object
+ */
+function handleSavedClientID(mutableState: GlobalContextType, clientID: string) {
+    mutableState.clientID = clientID
+    return mutableState
+}
+
+/**
+ * Sets the client ID saved in state to be the controlling client ID
+ * 
+ * @param {GlobalContextType} mutableState State object ref for making modifications
+ * @returns {GlobalContextType} Updated mutable state object
+ */
+function handleTakeControlSuccess(mutableState: GlobalContextType) {
+    mutableState.controllingClientID = mutableState.clientID
+    return mutableState
+}
+
+/**
+ * Turns off RC Mode on the client-side
+ * 
+ * @param {GlobalContextType} mutableState State object ref for making modifications
+ * @returns {GlobalContextType} Updated mutable state object
+ */
+function handleExitedRCMode(mutableState: GlobalContextType) {
+    mutableState.isRCMode = false
+    return mutableState
+}
+
+/**
+ * Closes the HubDetails panel
+ * 
+ * @param {GlobalContextType} mutableState State object ref for making modifications
+ * @returns {GlobalContextType} Updated mutable state object
+ */
+function handleClosedHubDetails(mutableState: GlobalContextType) {
+    mutableState.showHubDetails = false
+    return mutableState
+}
+
+/**
+ * Handles the interplay between selecting the hub and clicking the HubTab
+ * 
+ * @param {GlobalContextType} mutableState State object ref for making modifications
+ * @returns {GlobalContextType} Updated mutable state object
+ */
+function handleClickedHubTab(mutableState: GlobalContextType) {
+    const isHubSelected = mutableState.selectedPodElement !== null && mutableState.selectedPodElement.type === PodElement.HUB
+
+    if (isHubSelected) {
+        mutableState.showHubDetails = false
+    } else {
+        mutableState.showHubDetails = true
+    }
+
+    if (mutableState.showHubDetails) {
+        mutableState.selectedPodElement = { type: PodElement.HUB, id: 1 }
+    } else {
+        mutableState.selectedPodElement = null
+    }
+
+    return mutableState
+}
+
+/**
+ * Handles the interplay between selecting a bot and clicking a BotTab
+ * 
+ * @param {GlobalContextType} mutableState State object ref for making modifications
+ * @returns {GlobalContextType} Updated mutable state object
+ * 
+ * @notes
+ * This function only unselects the hub and closes the HubDetails panel. 
+ * It does not handle bot selection yet.
+ */
+function handleClickedBotTab(mutableState: GlobalContextType) {
+    if (mutableState.selectedPodElement !== null && mutableState.selectedPodElement.type === PodElement.HUB) {
```

Comment:
Can we be consistent with this conditional in terms of setup. See above function.